### DISCUSSION TITLE
Mouse speed option for scrolling

### DISCUSSION THREAD

**Discussion by MachoRock** [sumatrapdfreader/sumatrapdf#2571] [Ideas]
Hi all, I have an idea. How to add mouse speed option for scrolling, so we can read manga in any zoom level while can adjust different speed for mouse scrolling. I think this is useful for enthusiast reader.

**Comment by GitHubRulesOK** [comment]
To increase mouse steps to cat size steps you can use **"press Alt while scrolling : scrolls faster"** *NOTE HOLD DOWN ALT BEFORE SCROLL* or alter mouse steps in windows mouse settings.
but if you use a  wheel with middle screen click switch it should be click to pan (in all directions not simple up down) at a speed that's set by distance from mid page click point.

**Reply by NolanDeveloper** [reply]
Alt key triggers top bar menu.

**Reply by GitHubRulesOK** [reply]
@NolanDeveloper thanks I have corrected the omission to mention it is only WHILE scrolling

**Reply by NolanDeveloper** [reply]
It still opens the menu even if I scroll

**Reply by GitHubRulesOK** [reply]
@NolanDeveloper You are correct 
That comment is less than clear the actual function is HOLD alt without release then scroll !! so have added that too

**Reply by NolanDeveloper** [reply]
I'm probably missing something obvious. Behavior for me is such: I hold alt, scroll (it scrolls faster), release alt, menu opens.

**Reply by GitHubRulesOK** [reply]
@NolanDeveloper Hmm as said different users hardware setting different behaviours (I am using pre-release on Windows 10)
so perhaps your version or windows is adding some extra press or sticky keys ,since for me:
Alt menu function needs 2 "alts" to *kick in after scroll* as normal function Windows Menu Systems trigger which is "global"(normally will only need one alt).

**Comment by ghost** [comment]
The Alt tip is nice, but the downside is that you have to keep pressing it. Any chances that could be locked or add an option for how many lines normal scrolling does?

**Reply by GitHubRulesOK** [reply]
my best suggestion is find an app that is app aware such as AutoHotKey or a mouse controller like ["Highresolution Enterprises" X-mouse](https://www.highrez.co.uk/downloads/XMouseButtonControl.htm) (Providing up to 10 layers of button mapping for every profile, Per-Profile Based Scrolling), however messing with mouse scroll is not easy, as driven by different hardware drivers Mouse/Touch screen/Touch pad/JoyStick/Eye tracker etc.. Hence my preference is Auto scroll

**Comment by jsteave** [comment]
@[MachoRock ](https://malloftoys.com/collections/lui)
Do you know how to quickly configure your mouse to scroll from 1 line to 3 at a time?

I already know how to do it through the mouse control panel but I'd like a faster (lazier) method than fishing through menus and control panels

I suppose I could find out what registry key configures that and write a .reg file but being retired I'd really rather not

My reason is scrolling 3 lines is best for web browsers but gaming is best with 1 line scrolling

**Reply by GitHubRulesOK** [reply]
To make the changes done by the Registry tweak take effect, you need to [sign out](https://winaero.com/blog/all-ways-to-sign-out-from-windows-10/) and sign in to your user account.

don't think that's avoidable, but see below
https://duckduckgo.com/?q=mouse+scroll+registry+reg&ia=web
 Computer\HKEY_CURRENT_USER\Control Panel\Mouse

So not that simple as its a hardware setting at startup so scripting user control panel run time settings is easier
however see https://stackoverflow.com/questions/17040534/how-to-change-mouse-scroll-speed-using-batch-file-in-windows

**Comment by NolanDeveloper** [comment]
The way it's implemented at the current moment is a bug. I have to use my finger 3-4 times to scroll 1 page. It's too slow. No other app scrolls this slow.

**Reply by GitHubRulesOK** [reply]
@NolanDeveloper 
I am not the developer but AFAIK the issue is "different windows (users) different strokes"
 SumatraPDF is a wrapper around MuPDF and Windows Hardware is a Wrapper around SumatraPDF UX
 so SumatraPDF can **try** to control the speed of MuPDF renders
 but it is Windows controls the speed of SumatraPDF and thus may differ from mouse to mouse etc.

As an example the Microsoft brand mouse I am using now will one scroll click jump this page this much at a time at 100% in edge 
<img width="114" height="99" alt="image" src="https://github.com/user-attachments/assets/f5b1486c-e4c1-4e0b-9987-8056e18fb930" /> and at 200% only this much 
<img width="228" height="102" alt="image" src="https://github.com/user-attachments/assets/a6df79bc-9e3d-4d8e-b89d-6bab98088151" />
which is the same as one click at 100% in SumatraPDF Thus Sumatra PDF is 2 times as many clicks as Edge and many complaints that is not slow enough. but with acceleration one click rolls this much 
<img width="124" height="252" alt="image" src="https://github.com/user-attachments/assets/dced9e64-dcf3-437f-ae52-027d16c72703" />
so is considerably faster

**Reply by NolanDeveloper** [reply]
@GitHubRulesOK From my experience the scroll behavior of SumatraPDF differs significantly from other windows applications. I have used SumatraPDF in various configurations over the years, on different computers and with different mice, and the scrolling has consistently been noticeably slower than in other applications. The question is whether this is a general issue rather than an isolated case. The specific implementation details of SumatraPDF are not relevant to determining whether the issue exists.

**Reply by kjk** [reply]
There are many ways to scroll, so which method specifically is too slow?

BTW: I've made changed to scroll speed in pre-release, so did you test with latest version?

**Reply by NolanDeveloper** [reply]
@kjk The issue is with mouse wheel scrolling although arrow scrolling is also very slow. I'm now currently at version 3.3.3 and have just confirmed the issue: in chrome, telegram and file explorer mouse wheel scrolling has roughly the same speed which for me is 1 page per full range of finger motion. In Sumatra it's roughly twice as slow.

**Reply by kjk** [reply]
Well, try https://www.sumatrapdfreader.org/prerelease  this changed since 3.3.3
